tests: failing test pipeline in test_nodal_clearing.py due to xarray version 2026.02.0

Currently, our test pipeline is failing due to the dependency from pypsa -> linopy -> xarray. 
xarray 2026.04.0 was released two days ago and now throws an error: https://docs.xarray.dev/en/stable/whats-new.html#breaking-changes

Proposed fixes:
- Pin xarray down to 2026.02.0
- Wait for upstream fix to solve this issue

The issue is this:

```python
tests/test_nodal_clearing.py:377: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
assume/markets/clearing_algorithms/nodal_clearing.py:296: in clear
    status, termination_condition = n.optimize(
../../../.conda/envs/assume-new/lib/python3.14/site-packages/pypsa/optimization/optimize.py:527: in __call__
    m = n.optimize.create_model(
../../../.conda/envs/assume-new/lib/python3.14/site-packages/pypsa/optimization/optimize.py:634: in create_model
    n._model = Model(**kwargs)
               ^^^^^^^^^^^^^^^
../../../.conda/envs/assume-new/lib/python3.14/site-packages/linopy/model.py:181: in __init__
    self._objective: Objective = Objective(LinearExpression(None, self), self)
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
../../../.conda/envs/assume-new/lib/python3.14/site-packages/linopy/expressions.py:384: in __init__
    data = Dataset(data.transpose(..., TERM_DIM))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[RecursionError('maximum recursion depth exceeded') raised in repr()] Dataset object at 0x7f50cebbb750>
data_vars = <xarray.Dataset> Size: 8B
Dimensions:  (_term: 0)
Dimensions without coordinates: _term
Data variables:
    const    float64 8B 0.0
    coeffs   (_term) float64 0B 
    vars     (_term) int64 0B 
coords = None, attrs = None

    def __init__(
        self,
        # could make a VariableArgs to use more generally, and refine these
        # categories
        data_vars: DataVars | None = None,
        coords: Mapping[Any, Any] | None = None,
        attrs: Mapping[Any, Any] | None = None,
    ) -> None:
        if data_vars is None:
            data_vars = {}
        if isinstance(data_vars, Dataset):
>           raise TypeError(
                "Passing a Dataset as `data_vars` to the Dataset constructor is"
                " not supported. Use `ds.copy()` to create a copy of a Dataset."
            )
E           TypeError: Passing a Dataset as `data_vars` to the Dataset constructor is not supported. Use `ds.copy()` to create a copy of a Dataset.
```

pinning "xarray <v2026.04.0" helps.

--- a/docs/source/release_notes.rst
+++ b/docs/source/release_notes.rst
@@ -16,6 +16,8 @@ Upcoming Release
 **New Features:**
   - **Generic Forecasting Interface**: This interface enables to specify different forecast algorithms for preprocess, initialization and update during runtime. They can be specified in the config.yaml or unit csv files. For more information about currently implemented algorithms and how to specify them please read the documentation on Unit forecasts.
 
+**Bug Fixes:**
+  - **dependencies**: pin xarray and setuptools dependencies until upstream fixes are available
 
 0.6.0 - (18th March 2026)
 =========================

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -43,7 +43,7 @@ dependencies = [
     "pyyaml >=6.0",
     "pyyaml-include >=2.2a",
     "pyomo >=6.8.0",
-    "xarray",
+    "xarray <v2026.04.0", # https://github.com/assume-framework/assume/issues/790
     "highspy",
 ]
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -51,6 +51,7 @@ dependencies = [
 learning = [
     "torch >=2.0.1",
     "tensorboard >=2.7.0",
+    "setuptools <82" # https://github.com/tensorflow/tensorboard/issues/7003
 ]
 network = [
     "pypsa",

